File: .github/scripts/validate-config.py
#!/usr/bin/env python3
"""
Configuration validation script for GitHub Actions workflow.
Validates package.json and tsconfig.json files.
"""

import json
import sys
import os

def validate_package_json():
    """Validate package.json file."""
    try:
        with open('logistics-lynx/package.json') as f:
            data = json.load(f)
        
        # Check required fields
        required_fields = ['name', 'version', 'scripts']
        for field in required_fields:
            if field not in data:
                print(f'❌ Missing required field: {field}')
                return False
        
        # Check required scripts
        scripts = data.get('scripts', {})
        required_scripts = ['build', 'lint']
        optional_scripts = ['test', 'dev', 'start', 'deploy']
        
        for script in required_scripts:
            if script in scripts:
                print(f'✅ {script} script present')
            else:
                print(f'❌ Missing required script: {script}')
                return False
        
        for script in optional_scripts:
            if script in scripts:
                print(f'✅ {script} script present (optional)')
            else:
                print(f'⚠️ {script} script missing (optional)')
        
        print('✅ package.json validation passed')
        return True
    except Exception as e:
        print(f'❌ package.json validation failed: {e}')
        return False

def validate_tsconfig_json():
    """Validate tsconfig.json file."""
    try:
        with open('logistics-lynx/tsconfig.json') as f:
            content = f.read()
        
        # Remove TypeScript-style comments before parsing as JSON
        import re
        
        # Remove single-line comments (// ...)
        content = re.sub(r'//.*$', '', content, flags=re.MULTILINE)
        
        # Remove multi-line comments (/* ... */)
        content = re.sub(r'/\*.*?\*/', '', content, flags=re.DOTALL)
        
        # Parse the cleaned JSON
        data = json.loads(content)
        
        required_fields = ['compilerOptions']
        for field in required_fields:
            if field not in data:
                print(f'❌ Missing required field: {field}')
                return False
        
        # Check for essential compiler options
        compiler_options = data.get('compilerOptions', {})
        essential_options = ['target', 'module', 'moduleResolution']
        
        for option in essential_options:
            if option in compiler_options:
                print(f'✅ {option} compiler option present')
            else:
                print(f'⚠️ {option} compiler option missing (optional)')
        
        print('✅ tsconfig.json validation passed')
        return True
    except Exception as e:
        print(f'❌ tsconfig.json validation failed: {e}')
        return False

def main():
    """Main validation function."""
    print("⚙️ Validating configuration files...")
    
    success = True
    
    # Validate package.json
    if os.path.exists('logistics-lynx/package.json'):
        print("📦 Validating logistics-lynx/package.json...")
        if not validate_package_json():
            success = False
    else:
        print("❌ logistics-lynx/package.json not found")
        success = False
    
    # Validate TypeScript config
    if os.path.exists('logistics-lynx/tsconfig.json'):
        print("📝 Validating tsconfig.json...")
        if not validate_tsconfig_json():
            success = False
    else:
        print("❌ logistics-lynx/tsconfig.json not found")
        success = False
    
    if not success:
        sys.exit(1)
    
    print("✅ All configuration files validated successfully")

if __name__ == "__main__":
    main()

File: .github/scripts/validate-config.js
#!/usr/bin/env node
/**
 * Configuration validation script for GitHub Actions workflow.
 * Validates package.json and tsconfig.json files.
 */

import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

function validatePackageJson() {
  try {
    const packagePath = path.join(__dirname, '../../logistics-lynx/package.json');
    if (!fs.existsSync(packagePath)) {
      console.log(`❌ File not found: ${packagePath}`);
      return false;
    }
    const data = JSON.parse(fs.readFileSync(packagePath, 'utf8'));
    
    // Check required fields
    const requiredFields = ['name', 'version', 'scripts'];
    for (const field of requiredFields) {
      if (!(field in data)) {
        console.log(`❌ Missing required field: ${field}`);
        return false;
      }
    }
    
    // Check required scripts
    const scripts = data.scripts || {};
    const requiredScripts = ['build', 'lint'];
    const optionalScripts = ['test', 'dev', 'start', 'deploy'];
    
    for (const script of requiredScripts) {
      if (script in scripts) {
        console.log(`✅ ${script} script present`);
      } else {
        console.log(`❌ Missing required script: ${script}`);
        return false;
      }
    }
    
    for (const script of optionalScripts) {
      if (script in scripts) {
        console.log(`✅ ${script} script present (optional)`);
      } else {
        console.log(`⚠️ ${script} script missing (optional)`);
      }
    }
    
    console.log('✅ package.json validation passed');
    return true;
  } catch (error) {
    console.log(`❌ package.json validation failed: ${error.message}`);
    return false;
  }
}

function validateTsConfigJson() {
  try {
    const tsConfigPath = path.join(__dirname, '../../logistics-lynx/tsconfig.json');
    if (!fs.existsSync(tsConfigPath)) {
      console.log(`❌ File not found: ${tsConfigPath}`);
      return false;
    }
    
    // Parse the JSON directly since we have a clean file
    const data = JSON.parse(fs.readFileSync(tsConfigPath, 'utf8'));
    
    const requiredFields = ['compilerOptions'];
    for (const field of requiredFields) {
      if (!(field in data)) {
        console.log(`❌ Missing required field: ${field}`);
        return false;
      }
    }
    
    // Check for essential compiler options
    const compilerOptions = data.compilerOptions || {};
    const essentialOptions = ['target', 'module', 'moduleResolution'];
    
    for (const option of essentialOptions) {
      if (option in compilerOptions) {
        console.log(`✅ ${option} compiler option present`);
      } else {
        console.log(`⚠️ ${option} compiler option missing (optional)`);
      }
    }
    
    console.log('✅ tsconfig.json validation passed');
    return true;
  } catch (error) {
    console.log(`❌ tsconfig.json validation failed: ${error.message}`);
    return false;
  }
}

function main() {
  console.log("⚙️ Validating configuration files...");
  
  let success = true;
  
  // Validate package.json
  const packagePath = path.join(__dirname, '../../logistics-lynx/package.json');
  if (fs.existsSync(packagePath)) {
    console.log("📦 Validating logistics-lynx/package.json...");
    if (!validatePackageJson()) {
      success = false;
    }
  } else {
    console.log("❌ logistics-lynx/package.json not found");
    success = false;
  }
  
  // Validate TypeScript config
  const tsConfigPath = path.join(__dirname, '../../logistics-lynx/tsconfig.json');
  if (fs.existsSync(tsConfigPath)) {
    console.log("📝 Validating tsconfig.json...");
    if (!validateTsConfigJson()) {
      success = false;
    }
  } else {
    console.log("❌ logistics-lynx/tsconfig.json not found");
    success = false;
  }
  
  if (!success) {
    process.exit(1);
  }
  
  console.log("✅ All configuration files validated successfully");
}

// Execute main function
main();

export { validatePackageJson, validateTsConfigJson };
